prov/rxm/src/rxm_domain.c

```diff
@@ -404,6 +406,18 @@ static int rxm_domain_close(fid_t fid)
 	if (ret)
 		return ret;
 
+	if (rxm_domain->offload_coll_domain) {
+		ret = fi_close(&rxm_domain->offload_coll_domain->fid);
+		if (ret)
+			return ret;
```

Comment:
On success, set rxm_domain->offload_coll_domain = NULL.  This way, if the close call below fails and the app tries calling close again, we don't crash.

Reply:
Done.